flow/project.py

```diff
@@ -3575,18 +4065,33 @@ def _main_exec(self, args):
             except KeyError:
                 operation = self._operations[args.operation]
 
-                def operation_function(job):
-                    cmd = operation(job).format(job=job)
+                def operation_function(*jobs):
+                    cmd = operation(jobs).format(jobs=jobs)
                     subprocess.run(cmd, shell=True, check=True)
 
         except KeyError:
             raise KeyError("Unknown operation '{}'.".format(args.operation))
 
-        if getattr(operation_function, '_flow_aggregate', False):
-            operation_function(jobs)
-        else:
-            for job in jobs:
-                operation_function(job)
+        try:
+            filter = operation_function._flow_select
+            grouper, sort = operation_function._flow_aggregate
+        except AttributeError:
+            filter = operation.cmd._flow_select
+            grouper, sort = operation.cmd._flow_aggregate
+
+        jobs = list(jobs)
+        jobs_list = filter(jobs)
+        if sort is not None:
+            jobs_list = sort(jobs_list)
+        jobs_list = grouper([job for job in jobs_list])
+
+        for job_list in jobs_list:
+            job_list = list(job_list)
+            for i, job in enumerate(job_list):
+                if job is None:
+                    del job_list[i:]
+                    break
```

Comment:
This code is repeated frequently. We should combine this logic into a function or class.

Reply:
I'll merge this concept with the `aggregate` class.